pandas/core/generic.py

```diff
@@ -2910,9 +2910,11 @@ def to_csv(self, path_or_buf=None, sep=",", na_rep='', float_format=None,
 
         index : bool, default True
             Write row names (index).
-        index_label : str or sequence, or False, default None
+        index_label : False or str or sequence, default None
             Column label for index column(s) if desired. If None is given, and
-            `header` and `index` are True, then the index names are used. A
+            `header` and `index` are True, then the index names are used. If
+            `header` is False or None, default index_label changes to False,
```

Comment:
Thanks for the review, @jreback ! Can you say more about it? The change now is because, in previous api design, when `header` is set to None or False, col names and index names will be all gone (won't be printed), and in this new change, the index name and col names are separated, and default `index_label` is `None` which means index names will still be printed if they exist. Therefore, following @simonjayhawkins suggestion to maintain backwards compatibility, in our internal `CSVFormatter`, the index_label will be changed to `False` given `header` is `None` or `False`. And I am glad to hear more feedbacks!

Reply:
sorry if i was not clear..

i was think along the lines

```
index_label : str or sequence, or bool, default { True,
                                                  False if header or index are False
```

this should result in an non breaking api and maintain the same behaviour for `None` which does not need to be included in the docstring

Reply:
thanks! i just changed and removed None! @simonjayhawkins

Reply:
oh, wait!! would this still not be able to display when `header=False` since `index_label` is coerced to `False` by default? @simonjayhawkins

Reply:
if its explicitly set to `True` by the caller then the default values become irrelevant.